src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5732,6 +5759,24 @@ static QVariant fcnLineLocatePoint( const QVariantList &values, const QgsExpress
   return distance >= 0 ? distance : QVariant();
 }
 
+static QVariant fcnLineLocateM( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry lineGeom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  const double m = QgsExpressionUtils::getDoubleValue( values.at( 1 ), parent );
+  const bool use3DDistance = values.at( 2 ).toBool();
+
+  double x, y, z, distance;
+
+  const QgsLineString *line = qgsgeometry_cast<const QgsLineString *>( lineGeom.constGet() );
+  if ( !line )
+  {
+    return QVariant();
```

Comment:
error eval message?

Reply:
I think we should match the behavior to the pre-existing line_locate_point() which does not return an evaluation error string.